key.js

```diff
@@ -1,18 +1,321 @@
+    }
+    function gameMusicPlay() {
+        const backgroundMusic = gameState.gameAudio.getGameAudio();
+        backgroundMusic.currentTime = 0;
+        backgroundMusic.play();
+    }
+    function gameMusicStop() {
+        const backgroundMusic = gameState.gameAudio.getGameAudio();
+        backgroundMusic.pause();
+    }
+    function muteAudio() {
+        gameState.sound = false;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds(),
+            keyboardSounds.getInvalidAudio(),
+            keyboardSounds.getGreenAudio(),
+            keyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => audio.pause());
+    }
+    function unMuteAudio() {
+        gameState.sound = true;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds(),
+            keyboardSounds.getInvalidAudio(),
+            keyboardSounds.getGreenAudio(),
+            keyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => {
+            audio.currentTime = 0;
+            audio.play()
+        });
+    }
+    function decrementGameLife() {
+        gameState.lives--;
+    }
+
+    function printGameState() {
+        console.log(gameState);
+    }
+
+    return {
+        startGame,
+        gameOver,
+        levelUp,
+        resetGameLivesAndTimer,
+        resetGameState,
+        muteAudio,
+        unMuteAudio,
+        decrementGameLife,
+        printGameState,
+        getGameLife,
+        endGame,
+        gameMusicPlay,
+        gameMusicStop
+    }
+    // I. create a game state object (ensure it's a deep copy of initialGameState)
+    // II. create utility methods to manipulate the game state object
+    // 1. incrementGameLevel
+    // 2. getGameLevel
+    // 3. startGame
+    // 4. endGame
+    // 5. isGameRunning
+    // 6. isGameOver
+    // 7. getGameLife
+    // 8. decrementGameLife
+    // 9. resetGameState
+    // 10. getGameAudio
+    // 11. printGameState // just for debugging
+    // 12. <create any other method you feel that makes sense>
+    // III. return an object with all the aforementioned utility methods from 1..9.
+
+    // return {
+    //     incrementGameLevel: () => gameState.levelsInfo.currentLevel++
+    // }
+})();
+// IV. Refactor the code below to use the gameModule above for creating the game logic.
 // Button Animation Logic :
 
+const viewModule = (function () {
+    const DOMStrings = {
+        entireContainer: 'container',
+        changeHeading: 'main_heading',
+        defaultLives: 'default',
+        currentLives: 'lives',
+        mainButton: 'main',
+        liveDisplay: 'heart',
+        timerDisplay: 'timer',
+        keyDirections: 'btns',
+        randomKeyArrows: 'arrows',
+        keyboardContents: 'contents'
```

Comment:
Naming is the most important aspect of writing good programs.
Please don't use different element names for HTML class elements, and the ones in the JS code.

Please change the naming to match whatever exists in the frontend (HTML/CSS) to whatever variable is named in the script (JS), or vice-versa.

For example, you can change the naming of the HTML/CSS class elements' names to whatever are defined in the script as `DOMStrings` here.

```suggestion
        currentLives: 'current_lives',
        mainButton: 'main_button',
        livesDisplay: 'lives_display',
        timerDisplay: 'timer_display',
        keyDirections: 'key_directions',
        randomKeyArrows: 'random_key_arrows',
        keyboardContents: 'keyboard_contents'
```